src/Compilers/CSharp/Portable/Symbols/ConstantValueUtils.cs

```diff
@@ -155,5 +156,30 @@ internal static string UnescapeInterpolatedStringLiteral(string s)
 
             return value;
         }
+
+        private sealed class CheckConstantInterpolatedStringValidity : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+        {
+            internal readonly DiagnosticBag diagnostics;
+
+            public CheckConstantInterpolatedStringValidity(DiagnosticBag diagnostics)
+            {
+                this.diagnostics = diagnostics;
+            }
+
+            public override BoundNode VisitInterpolatedString(BoundInterpolatedString node)
+            {
+                Binder.CheckFeatureAvailability(node.Syntax, MessageID.IDS_FeatureConstantInterpolatedStrings, diagnostics);
+                return null;
+            }
+        }
+
+        internal static void CheckLangVersionForConstantInterpolatedStrings(BoundExpression expression, DiagnosticBag diagnostics)
```

Comment:
>CheckLangVersionForConstantInterpolatedStrings [](start = 29, length = 46)

I would be less specific here. Just `CheckLangVersionForConstantValue`. That way, if we do anything else in this area in the future with non-strings, we have an obvious place to put it.